runtime/lua/vim/diagnostic.lua

```diff
@@ -1169,15 +1172,21 @@ end
 ---                      highlight.
 ---                      Overrides the setting from |vim.diagnostic.config()|.
 ---@return tuple ({float_bufnr}, {win_id})
-function M.open_float(bufnr, opts)
-  vim.validate {
-    bufnr = { bufnr, 'n', true },
-    opts = { opts, 't', true },
-  }
+function M.open_float(opts, ...)
+  -- Support old (bufnr, opts) signature
+  local bufnr
+  if opts == nil or type(opts) == "number" then
+    bufnr = opts
+    opts = ...
+  else
+    vim.validate {
+      opts = { opts, 't', true },
+    }
+  end
 
   opts = opts or {}
-  bufnr = get_bufnr(bufnr)
-  local scope = opts.scope or "line"
+  bufnr = get_bufnr(bufnr or opts.bufnr)
```

Comment:
should this be a general enum somwhere?

Reply:
enums can't be scoped globally so it would have to be `scope = vim.diagnostic.scope.LINE` or something, I'm not sure that that's better.